base/applications/cmdutils/fc/fc.c

```diff
@@ -0,0 +1,498 @@
+    CloseHandle(hFile2);
+    return ret;
+}
+
+static FCRET
+UnicodeTextCompare(const FILECOMPARE *pFC, HANDLE hMapping1, size_t cb1,
+                                           HANDLE hMapping2, size_t cb2)
+{
+    FCRET ret;
+    BOOL fIgnoreCase = !!(pFC->dwFlags & FLAG_C);
+    DWORD dwCmpFlags = (fIgnoreCase ? NORM_IGNORECASE : 0);
+    LPWSTR psz1, psz2;
+    size_t ib = 0, cch1 = cb1 / sizeof(WCHAR), cch2 = cb2 / sizeof(WCHAR);
+
+    do
+    {
+        psz1 = MapViewOfFile(hMapping1, FILE_MAP_READ, HILONG(ib), LOLONG(ib), (DWORD)cb1);
+        psz2 = MapViewOfFile(hMapping2, FILE_MAP_READ, HILONG(ib), LOLONG(ib), (DWORD)cb2);
+        if (!psz1 || !psz2)
+        {
+            ret = OutOfMemory();
+            break;
+        }
+        if (CompareStringW(0, dwCmpFlags, psz1, (INT)cch1, psz2, (INT)cch2) == CSTR_EQUAL)
+        {
+            ret = NoDifference();
+            break;
+        }
+        // TODO: compare each lines
+        // TODO: large file support
+        ret = Different(pFC->file1, pFC->file2);
+    } while (0);
+
+    UnmapViewOfFile(psz1);
+    UnmapViewOfFile(psz2);
+    return ret;
+}
+
+static FCRET
+AnsiTextCompare(const FILECOMPARE *pFC, HANDLE hMapping1, size_t cb1,
+                                        HANDLE hMapping2, size_t cb2)
+{
+    FCRET ret;
+    BOOL fIgnoreCase = !!(pFC->dwFlags & FLAG_C);
+    DWORD dwCmpFlags = (fIgnoreCase ? NORM_IGNORECASE : 0);
+    LPSTR psz1, psz2;
+    size_t ib = 0;
+
+    do
+    {
+        psz1 = MapViewOfFile(hMapping1, FILE_MAP_READ, HILONG(ib), LOLONG(ib), (DWORD)cb1);
+        psz2 = MapViewOfFile(hMapping2, FILE_MAP_READ, HILONG(ib), LOLONG(ib), (DWORD)cb2);
+        if (!psz1 || !psz2)
+        {
+            ret = OutOfMemory();
+            break;
+        }
+        if (CompareStringA(0, dwCmpFlags, psz1, (INT)cb1, psz2, (INT)cb2) == CSTR_EQUAL)
+        {
+            ret = NoDifference();
+            break;
+        }
+        // TODO: compare each lines
+        // TODO: large file support
+        ret = Different(pFC->file1, pFC->file2);
+    } while (0);
+
+    UnmapViewOfFile(psz1);
+    UnmapViewOfFile(psz2);
+    return ret;
+}
+
+static FCRET TextFileCompare(const FILECOMPARE *pFC)
+{
+    FCRET ret;
+    HANDLE hFile1, hFile2, hMapping1 = NULL, hMapping2 = NULL;
+    size_t cb1, cb2;
+    BOOL fUnicode = !!(pFC->dwFlags & FLAG_U);
+
+    hFile1 = DoOpenFileForInput(pFC->file1);
+    if (hFile1 == INVALID_HANDLE_VALUE)
+        return FCRET_CANT_FIND;
+    hFile2 = DoOpenFileForInput(pFC->file2);
+    if (hFile2 == INVALID_HANDLE_VALUE)
+    {
+        CloseHandle(hFile1);
+        return FCRET_CANT_FIND;
+    }
+
+    do
+    {
+        if (lstrcmpiW(pFC->file1, pFC->file2) == 0)
+        {
+            ret = NoDifference();
+            break;
+        }
+        if (!GetFileSizeDx(hFile1, &cb1, FALSE))
+        {
+            ret = CannotRead(pFC->file1);
+            break;
+        }
+        if (!GetFileSizeDx(hFile2, &cb2, FALSE))
```

Comment:
And why large text files wouldn't be allowed?

Reply:
Because `CompareString` function won't allow `size_t` or `DWORDLONG`.

Reply:
Well then just do the comparison by batches ....